src/message_interface/message_handler.rs

```diff
@@ -119,61 +106,78 @@ impl WalletMessageHandler {
                         .await?;
                     let mut accounts = Vec::new();
                     for account_handle in account_handles {
                         let account = account_handle.read().await;
                         accounts.push(account.clone());
                     }
-                    Ok(ResponseType::ReadAccounts(accounts))
+                    Ok(Response::Accounts(accounts))
                 })
                 .await
             }
+            #[cfg(feature = "stronghold")]
+            MessageType::RestoreBackup { source, password } => {
+                convert_async_panics(|| async { self.restore_backup(source.to_path_buf(), password).await }).await
+            }
             #[cfg(feature = "storage")]
             MessageType::DeleteStorage => {
                 convert_async_panics(|| async {
                     self.account_manager.delete_storage().await?;
-                    Ok(ResponseType::DeletedStorage)
+                    Ok(Response::Ok(()))
                 })
                 .await
             }
             MessageType::GenerateMnemonic => convert_panics(|| {
                 self.account_manager
                     .generate_mnemonic()
-                    .map(ResponseType::GeneratedMnemonic)
+                    .map(Response::GeneratedMnemonic)
             }),
-            MessageType::VerifyMnemonic(mnemonic) => convert_panics(|| {
-                self.account_manager
-                    .verify_mnemonic(mnemonic)
-                    .map(|_| ResponseType::VerifiedMnemonic)
+            MessageType::VerifyMnemonic(mut mnemonic) => convert_panics(|| {
+                self.account_manager.verify_mnemonic(&mnemonic)?;
+                mnemonic.zeroize();
+                Ok(Response::Ok(()))
             }),
             MessageType::SetClientOptions(options) => {
                 convert_async_panics(|| async {
                     self.account_manager.set_client_options(*options.clone()).await?;
-                    Ok(ResponseType::UpdatedAllClientOptions)
+                    Ok(Response::Ok(()))
                 })
                 .await
             }
             MessageType::GetNodeInfo => {
+                convert_async_panics(|| async { self.account_manager.get_node_info().await.map(Response::NodeInfo) })
+                    .await
+            }
+            MessageType::SetStrongholdPassword(mut password) => {
+                convert_async_panics(|| async {
+                    self.account_manager.set_stronghold_password(&password).await?;
+                    password.zeroize();
+                    Ok(Response::Ok(()))
+                })
+                .await
+            }
+            MessageType::StoreMnemonic(mnemonic) => {
                 convert_async_panics(|| async {
-                    self.account_manager.get_node_info().await.map(ResponseType::NodeInfo)
+                    self.account_manager.store_mnemonic(mnemonic.to_string()).await?;
```

Comment:
Does it need to be zeroized ?

Reply:
Yes it should, will update

Reply:
Changed it so it gets moved inside, we need to add it to iota.rs there

Reply:
> Changed it so it gets moved inside, we need to add it to iota.rs there

Maybe create an issue so we don't forget about this.

Reply:
Done here https://github.com/iotaledger/iota.rs/issues/988